src/main/java/org/terasology/metalrenegades/ai/system/CitizenPersonalitySystem.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.metalrenegades.ai.system;
+
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.nameTags.NameTagComponent;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.metalrenegades.ai.event.CitizenSpawnedEvent;
+import org.terasology.namegenerator.creature.CreatureAssetTheme;
+import org.terasology.namegenerator.creature.CreatureNameProvider;
+import org.terasology.rendering.logic.SkeletalMeshComponent;
+import org.terasology.rendering.nui.Color;
+import org.terasology.utilities.random.FastRandom;
+import org.terasology.utilities.random.Random;
+
+/**
+ * Randomizes particular characteristics for city NPC characters.
+ */
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class CitizenPersonalitySystem extends BaseComponentSystem {
+
+    /**
+     * The +- range of possible model scaling differences.
+     */
+    public static final float SCALE_RANGE = 0.1f;
```

Comment:
Is a deviation of 0.1 noticable in-game? Nice idea to actually differ them in size, though 👍

Reply:
This has changed now since character scale changes are done by multiplying model scale, but this offset range allows for scale differences from 0.15f-0.35f, pretty significant on the small scale of the gooey model